File: 35.search-insert-position.java
import java.util.Arrays;

/*
 * @lc app=leetcode id=35 lang=java
 *
 * [35] Search Insert Position
 */

// @lc code=start
class Solution {
    public int searchInsert(int[] nums, int target) {
        int n = (int) Math.floor(nums.length / 2);
        while (true) {
            if (nums[n] == target) {
                break;
            } else if (nums.length == 1) {
                n++;
                break;
            } else if (nums[n] > target) {
                return searchInsert(Arrays.copyOfRange(nums, 0, n - 1), target);
            } else if (nums[n] < target) {
                return searchInsert(Arrays.copyOfRange(nums, n, nums.length - 1), target);
            }
        }
        return (int) n;
    }
}
// @lc code=end

File: 35.search-insert-position.py
#
# @lc app=leetcode id=35 lang=python
#
# [35] Search Insert Position
#

# @lc code=start
class Solution(object):
    def searchInsert(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        for idx, num in enumerate(nums):
            if num < target:
                pass
            else:
                return idx
        return len(nums)
        
        
# @lc code=end

